runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java

```diff
@@ -46,117 +51,216 @@
-	private final JLabel smithingLabel = new JLabel("--");
-	private final JLabel defenceLabel = new JLabel("--");
-	private final JLabel herbloreLabel = new JLabel("--");
-	private final JLabel fishingLabel = new JLabel("--");
-	private final JLabel rangedLabel = new JLabel("--");
-	private final JLabel thievingLabel = new JLabel("--");
-	private final JLabel cookingLabel = new JLabel("--");
-	private final JLabel prayerLabel = new JLabel("--");
-	private final JLabel craftingLabel = new JLabel("--");
-	private final JLabel firemakingLabel = new JLabel("--");
-	private final JLabel magicLabel = new JLabel("--");
-	private final JLabel fletchingLabel = new JLabel("--");
-	private final JLabel woodcuttingLabel = new JLabel("--");
-	private final JLabel runecraftLabel = new JLabel("--");
-	private final JLabel slayerLabel = new JLabel("--");
-	private final JLabel farmingLabel = new JLabel("--");
-	private final JLabel constructionLabel = new JLabel("--");
-	private final JLabel hunterLabel = new JLabel("--");
-	private final JLabel overallLabel = new JLabel("--");
-
-	private GridLayout stats;
+	private final IconTextField input;
+
+	private final List<JLabel> skillLabels = new LinkedList<>();
+
+	private final JPanel statsPanel = new JPanel();
+	private final JTextArea details = new JTextArea();
 
 	private final HiscoreClient client = new HiscoreClient();
+	private HiscoreResult result;
+	private CombatLevel combatLevel;
 
 	public HiscorePanel(RuneLite runelite)
 	{
 		this.runelite = runelite;
 
-		setMinimumSize(new Dimension(PANEL_WIDTH, PANEL_HEIGHT));
-		setPreferredSize(new Dimension(PANEL_WIDTH, PANEL_HEIGHT));
-		setSize(PANEL_WIDTH, PANEL_HEIGHT);
+		// Panel "constants"
+		// This was an EtchedBorder, but the style would change when the window was maximized.
+		Border subPanelBorder = BorderFactory.createLineBorder(this.getBackground().brighter(), 2);
+		Insets subPanelInsets = new Insets(2, 4, 2, 4);
+		Font labelFont = UIManager.getFont("Label.font");
+
+		// Setting base panel size
+		Dimension panelSize = new Dimension(PANEL_WIDTH, PANEL_HEIGHT);
+		setMinimumSize(panelSize);
+		setPreferredSize(panelSize);
+		setSize(panelSize);
 		setVisible(true);
 
-		input = new JTextField(11);
-		add(input);
+		// Create GBL to arrange sub items
+		GridBagLayout gridBag = new GridBagLayout();
+		setLayout(gridBag);
 
-		lookupButton = new JButton("Lookup");
-		add(lookupButton);
+		// Expand sub items to fit width of panel, align to top of panel
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.HORIZONTAL;
+		c.anchor = GridBagConstraints.NORTH;
 
-		lookupButton.addActionListener((ActionEvent e) ->
+		// Search box
+		JPanel inputPanel = new JPanel();
+		inputPanel.setLayout(new BorderLayout(7, 7));
+		inputPanel.setBorder(subPanelBorder);
+		Icon search = null;
+		try
+		{
+			search = new ImageIcon(ImageIO.read(HiscorePanel.class.getResourceAsStream("/search.png")));
```

Comment:
This should probably move under the resources of the plugin (net/runelite/client/plugins/hiscore)